packages/suite-messages/src/BaseTranslation.tsx

```diff
@@ -0,0 +1,94 @@
+    isNested?: boolean;
+} & ExtendedMessageDescriptor<TMessageId>;
+
+export type BaseTranslationProps<TMessageId extends string> = {
+    isReactNative?: boolean; // used to render span on the web, plain node for react-native
+    translationTooltip?: React.ComponentType<TooltipProps<TMessageId>>;
+    messages: MessagesObject<TMessageId>;
+} & TranslationProps<TMessageId>;
+
+export interface TooltipProps<TMessageId extends string> {
+    isNested?: boolean;
+    messageId: TMessageId;
+    children: any;
+}
+
+export type MessageValuesValue = string | number | boolean | Date | null | undefined;
+
+// Add MessageDescriptor type to values entry
+type FormatXMLElementFn = (...args: any[]) => string | object;
+export interface ExtendedMessageDescriptor<TMessageId extends string> extends MessageDescriptor {
+    id: TMessageId;
+    values?: {
+        [key: string]:
+            | MessageValuesValue
+            | React.ReactElement
+            | ExtendedMessageDescriptor<TMessageId>
+            | FormatXMLElementFn;
+    };
+}
+
+const isMsgType = <TMessageId extends string>(
+    props: BaseTranslationProps<TMessageId> | React.ReactNode,
+): props is BaseTranslationProps<TMessageId> =>
+    typeof props === 'object' &&
+    props !== null &&
+    (props as BaseTranslationProps<TMessageId>).id !== undefined;
+
+const RenderChildrenComponent = ({ children }: any) => children;
+
+export const BaseTranslation = <TMessageId extends string>(
+    props: BaseTranslationProps<TMessageId>,
+) => {
+    // Tooltip is only used in suite-landing-page, in case landing will be moved to eshop this could be removed
+    const TooltipComponent = props.translationTooltip ?? RenderChildrenComponent;
+    const values: Record<
+        string,
+        MessageValuesValue | React.ReactNode | ExtendedMessageDescriptor<TMessageId>
+    > = {};
+    // message passed via props (id, defaultMessage, values)
+    Object.keys(props.values || []).forEach(key => {
+        // Iterates through all values. The entry may also contain a MessageDescriptor.
+        // If so, Renders MessageDescriptor by passing it to `BaseTranslation` component
+        const maybeMsg = props.values![key];
+        values[key] = isMsgType(maybeMsg) ? (
+            <BaseTranslation<TMessageId>
+                {...maybeMsg}
+                translationTooltip={props.translationTooltip}
+                isNested
+                messages={props.messages}
+            />
+        ) : (
+            maybeMsg
+        );
+    });
+
+    // prevent runtime errors
+    if (
+        !props.defaultMessage &&
+        Object.prototype.hasOwnProperty.call(props, 'id') &&
+        !props.messages[props.id]
+    ) {
+        return <>{`Unknown translation id: ${props.id}`}</>;
+    }
+
+    const defaultTagName = props.isReactNative || props.isNested ? undefined : 'span';
+
+    return (
+        <TooltipComponent isNested={props.isNested} messageId={props.id}>
```

Comment:
I think this tooltip was only used [on the landing page](https://github.com/trezor/trezor-suite/blob/78736782fbd6556f536fcbc989ba15cf0b367555/packages/suite-web-landing/components/Translation/index.tsx#L21), maybe it's the right time to take it out of the base and somehow have it in `suite-web-landing` only?

P.S. Also [this](https://github.com/trezor/trezor-suite/blob/c297cad1b6031e140c6bc7b061ef1cdb52f15a0c/packages/suite/src/components/suite/Translation/index.tsx#L6) is not cool at all :(

Reply:
There is Translation component in landing which uses it `packages/suite-web-landing/components/Translation/index.tsx`

Reply:
btw. that "not cool" thing is replaced by default prop in BaseTranslation now. @dahaca

Reply:
True, good. Still this whole Tooltip code is a very legacy thing :(((